src/PluginsManager/Microsoft.Performance.Toolkit.PluginsManager.Core/Manager/IPluginsManager.cs

```diff
@@ -80,5 +85,146 @@ public interface IPluginsManager
+        ///     Whether to overwrite the existing plugin if there's one isntalled. 
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <param name="progress">
+        ///     Indicates the progress of plugin installation.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the plugin has been successfully installed. <c>false</c> otherwise.
+        /// </returns>
+        Task<bool> InstallLocalPluginAsync(
+           string pluginPackagePath,
+           bool overwriteInstalled,
+           CancellationToken cancellationToken,
+           IProgress<int> progress);
+
+        /// <summary>
+        ///     Uninstalls a plugin.
+        /// </summary>
+        /// <param name="installedPlugin">
+        ///     The plugin to be uninstalled.
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <param name="progress">
+        ///     Indicates the progress of plugin uninstallation.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the plugin has been successfully uninstalled. <c>false</c> otherwise.
+        /// </returns>
+        Task<bool> UninstallPluginAsync(
+            InstalledPlugin installedPlugin,
+            CancellationToken cancellationToken,
+            IProgress<int> progress);
+
+        /// <summary>
+        ///     Updates the plugin to another version.
+        /// </summary>
+        /// <param name="installedPlugin">
+        ///     The currently installed plugin.
+        /// </param>
+        /// <param name="targetPlugin">
+        ///     The available plugin to update to.
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <param name="progress">
+        ///     Indicates the progress of plugin uninstallation.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the plugin has been successfully updated. <c>false</c> otherwise.
+        /// </returns>
+        Task<bool> UpdatePluginAsync(
+            InstalledPlugin installedPlugin,
+            AvailablePlugin targetPlugin,
+            CancellationToken cancellationToken,
+            IProgress<int> progress);
+
+        /// <summary>
+        ///     Gets metadata of an installed plugin.
+        /// </summary>
+        /// <param name="installedPlugin">
+        ///     An installed plugin    
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <returns>
+        ///     Metadata of the given plugin.
+        /// </returns>
+        Task<PluginMetadata> GetInstalledPluginMetadataAsync(
```

Comment:
I would have thought installed plugins already have metadata available to get right off of it. Why do we need to hide this behind a method?